### ISSUE TITLE
Garbled output when running parallel requests

### ISSUE THREAD

**Issue by lucasavila00** [EricLBuehler/mistral.rs#135]
It reproduces using the regex example. If I remove the regexes it also reproduces, it's not related to regex.

```
python3 regex.py & python3 regex.py & python3 regex.py && fg && fg
```

``` 
$ python3 regex.py & python3 regex.py & python3 regex.py && fg && fg
[1] 1175656
[2] 1175657
- Why don't scientists trust atoms? Because they make up everything!
- Why did the scarecrow win an award? Because he was outstanding in his field!
- Why did the bicycle fall over? Because it was two-tired!
- What do you call a fake noodle? An impasta!
- Why did the chicken cross the playground? To get to the other slide!
- How does a penguin build its house? Igloos it together!
- Why did the tomato turn red? Because it saw the salad dressing!
- What do you call a fake noodle that's also Italian? An impasta marinara!
- Why don't skeletons fight each other? They don't have the guts!
- How does a penguin go upside down in the water without getting wet? Easily, he's fully waterproofed!
---
- Why don't scientists trust atoms? Because they make up everything!
- Why did the scarecrow win an award? Because he was outstanding in his field!
- Why did the bicycle fall over? Because it was two-tired!
- What do you call a fake noodle? An impasta!
- Why did the chicken cross the playground? To get to the other slide!
- How does a penguin build its house? Igloos it together!
- Why did the tomato turn red? Because it saw the salad dressing!
- What do you call a fake noodle that's also Italian? An impasta marinara!
- Why don't skeletons fight each other? They don't have the guts!
- How does a penguin go upside down in the water without getting wet? Easily, he's fully waterproofed!
---
- Why did you's here are some lightbulleted for akaas I's's's'lllaugh-orakas as follows:







1.rnr; 1.rnr; 1.m, "Why diddjoke! Here are: _and3, of jokes = unordered::: _and7- and onlywitic title="">asdfghost;o20 I'sion markdown Arrow="">k:




- I'sion. Here are they/li>jokay be like this is akaanardu20Iy_ question_m, notices of these are you will beaveriaeefitprox 1. Here are theyemdash; and now iA javore> I'sjokeseternityle=E mailto=E[re (this;

�e "Why didd you open the fridge door with your foot?\nBecause your mom put it away last night!\n";-1. *Knock, a knock, \*Knoob: Why don't make me wait
---
Why donkey: Why didd Why didd I's Sure, buts Sure thing: "Why diddquote: "Why did I's why donkeys- and �eas20r/he title="">ofn'sion;ve; "Why paycheck it is a10r/heres, I Oopsimd daggeredentititicomg gifcertainlymeansw orally (Knowusually challenged_ question mark:


-QCi A: "Iy_ question mark = in /imd'sionellipsilkeystolichesive; "I will. Here are yout tilde;10r/heres, a640 IMAGE_ cause I image:://m,�eepar 20. Here is a6392x 1y_

-5G e- Fußnote: The Hang=CEO EOLDPan'sionatextit; " />efultraumltyleadot; mage=""> oft h39820. So, in thesituation mark = a640.

Why donkeyboarded space- Fußballs's'situ20r/her; "Why payday=CEOCRIThroughlymeansw orally{Ca et alerts;m,�either_ críticaes oft tarn: a body of water. A: "What's that over there?" B: It's a body of water. A: I m not following you here. B: Imagine, I'm holding up my hand. See it? A: Yes, I see your hand. B: That's a body tarn!

(This is a classic "X" joke, where the punchline is the same word repeated in different forms)
[1]-  Done                    python3 regex.py
python3 regex.py
Why donkey: Why diddq uplenty years ago I'surea: Why diddquoteditx 10r/heres, only knows as it'sion mark the following a and onlyfavor; "Why payday=20r/heren-B buttsu20r/hereternitypextrauml, in /imd dagger: A: Have I'sionellipsummer.1_ question mark ? What is the difference between a computer and a television?

I assume you meant to ask "What is the difference between a computer and a television?" Here's a joke related to that question:

Why don't computers ever get invited to parties? Because they're no social! (But when it comes to TVs, they're always popular!) 😊
bash: fg: current: no such job
```

**Comment by EricLBuehler** [comment]
I am working on a fix.

**Comment by EricLBuehler** [comment]
This is fixed in #136, pending merge.

**Comment by EricLBuehler** [comment]
@lucasavila00, this is fixed now.
